### PR TITLE
use v2 api to set status without location

### PR DESCRIPTION
**Pull request by wow-such-code**
new samples in openbis are added to sample tracking using the new API, ignoring location and just setting status and timepoint

### CODE REVIEW

File: sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+
+import java.time.Instant
 
 interface SampleTrackingService {
 
     def registerFirstSampleLocation(String sampleCode, Location location) throws SampleUpdateException
 
+    def registerSampleStatus(String sampleCode, Status status, Instant timepoint) throws SampleUpdateException
+
 }
```
Comment (jenniferboedker) [inline sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy:14]:
```suggestion
}

```

File: sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/UpdateSampleStatus.groovy [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+import java.time.Instant
+
 
 interface UpdateSampleStatus {
 
     def updateSampleAtQBiC(String sampleCode) throws SampleUpdateException
 
+    def updateSample(String sampleCode, Instant timepoint) throws SampleUpdateException
+
 }
```
Comment (jenniferboedker) [inline sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/UpdateSampleStatus.groovy:12]:
```suggestion
}

```

File: sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy [hunk 3]
```diff
@@ -55,8 +59,22 @@ class SampleTrackingServiceConnector implements SampleTrackingService {
                 throw new SampleUpdateException("Could not update sample ${sampleCode}")
             }
         }
-
     }
 
+    @Override
+    def registerSampleStatus(String sampleCode, Status status, Instant timepoint) throws SampleUpdateException {
+        HttpClient client = RxHttpClient.create(service.rootUrl)
+        //TODO this is only a workaround, as the client seems not to prepend the base url.
```
Comment (jenniferboedker) [inline sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy:67]: what do you mean here?

File: sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
 import life.qbic.datamodel.samples.Location
+import life.qbic.datamodel.samples.Status
+
+import java.time.Instant
 
 interface SampleTrackingService {
 
     def registerFirstSampleLocation(String sampleCode, Location location) throws SampleUpdateException
 
+    def registerSampleStatus(String sampleCode, Status status, Instant timepoint) throws SampleUpdateException
```
Comment (jenniferboedker) [inline sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy:12]:
```suggestion
    def registerSampleWithFirstStatus(String sampleCode, Status status, Instant timepoint) throws SampleUpdateException
```

Comment (jenniferboedker) [reply sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy:12]: Do we need the status here? Isnt it automatically set by the system --> metadata received?

File: sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy (hunk 3 above)
Comment (jenniferboedker) [reply sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy:67]: ahhh 😅

Comment (jenniferboedker) [reply sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy:67]: is it the correct url?

Comment (jenniferboedker) [reply sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy:67]: @KochTobi is it how we want to continue doing it? If yes I vote for remove the todo for both methods

File: sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy (hunk 4 above)
Comment (jenniferboedker) [reply sample-status-updater-domain/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingService.groovy:12]: Ah now I see it. Sorry I was confused on how the classes work together. Makes sense then. So nothing to do here :+1:

File: sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy (hunk 3 above)
Comment (KochTobi) [reply sample-status-updater-infrastructure/src/main/groovy/life/qbic/samplestatusupdater/update/SampleTrackingServiceConnector.groovy:67]: Might be an outdated comment? I am unsure what it tells us so feel free to remove it.

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**